### PR TITLE
JIT: remove early prop's type propagation

### PR DESCRIPTION
**Pull request by AndyAyersMS**
This doesn't do any actual propagation, so remove it and the associated
tracking flags.

### CODE REVIEW

**Comment by AndyAyersMS** [comment]
cc @dotnet/jit-contrib 

No diffs per SPMI & crossgen PMI.

**Comment by erozenfeld** [comment]
> This doesn't do any actual propagation

Do you know why? Has something changed or has it never done any propagation?

File: src/coreclr/src/jit/earlyprop.cpp [hunk 1]
```diff
@@ -321,26 +319,7 @@ GenTree* Compiler::optEarlyPropRewriteTree(GenTree* tree, LocalNumberToNullCheck
...
-            // form of
-            //      *  stmtExpr  void  (top level)
-            //      \--*  indir     int
-            //          \--*  lclVar    ref    V02 loc0
-            if (compCurStmt->GetRootNode() == tree)
-            {
-                return nullptr;
-            }
-
-            objectRefPtr = tree->AsIndir()->Addr();
```
Comment (briansull) [inline src/coreclr/src/jit/earlyprop.cpp:337]: This tree looks like the old form of a null-check that was replaced by the GT_NULLCHECK node

**Comment by AndyAyersMS** [comment]
> Do you know why?

No, I don't.

It's possible this is now done earlier on, via type propagation in the importer.

**Comment by ViktorHofer** [comment]
// Auto-generated message

https://github.com/dotnet/runtime/commit/69e114c1abf91241a0eeecf1ecceab4711b8aa62 which was merged 12/7 removed the intermediate src/coreclr/**src**/ folder. This PR needs to be updated as it touches files in that directory which causes conflicts.

To update your commits you can use this bash script: https://gist.github.com/ViktorHofer/6d24f62abdcddb518b4966ead5ef3783. Feel free to use the comment section of the gist to improve the script for others.

**Comment by AndyAyersMS** [comment]
Rebased on top of src move.

**Comment by AndyAyersMS** [comment]
ping @dotnet/jit-contrib -- I have other changes that build on this one.

File: src/coreclr/jit/earlyprop.cpp [hunk 2]
```diff
@@ -4,7 +4,7 @@
 //                                    Early Value Propagation
 //
 // This phase performs an SSA-based value propagation optimization that currently only applies to array
-// lengths, runtime type handles, and explicit null checks. An SSA-based backwards tracking of local variables
+// lengths and explicit null checks. An SSA-based backwards tracking of local variables
 // is performed at each point of interest, e.g., an array length reference site, a method table reference site, or
```
Comment (BruceForstall) [inline src/coreclr/jit/earlyprop.cpp:8]: "a method table reference site," => also no longer applicable?

**Review by BruceForstall — APPROVED** [review]
LGTM

File: src/coreclr/jit/earlyprop.cpp (hunk 2 above)
Comment (AndyAyersMS) [reply src/coreclr/jit/earlyprop.cpp:8]: Right. Will fix that with my next PR.

**Pull request merged**